docs/intro-to-dune/index.md

```diff
@@ -0,0 +1,284 @@
+# Introduction to Dune
+
+Depending on how you've been following along, you may have several components in
+your project. Since these components don't have much in common with each other,
+it makes sense to put them in separate, independent apps. To do that, we'll have
+to dive into [Dune](https://dune.build/), a build system designed for OCaml
+projects.
+
```

Comment:
I miss the why on wanting to use dune:

- To organise your code properly and create different libraries
- To create a different arquitecture
- ...

Reply:
I added a sentence about why we use Dune a link to the features section in https://github.com/melange-re/melange-for-react-devs/pull/6/commits/be21ece850e3c046f1a94b4a0f2d6d3726693508